### PR TITLE
Fix bug with permissions part 2

### LINKED ISSUE #121: Process list shows processes within Category that shouldn't be visible by user.
**Describe the bug**
For user, on the list of processes are visible all processes - even if he/she hasn't got configured Read permission for some category.

**To Reproduce**
1. Having Flink and Nussknacker ran e.g. in docker with configured two categories: one visible for user A and the second not visible.
2. Open browser, login as a user A and go to 'main page'

**Expected behavior**
User A should see processes in categories that he/she has Read permission,
, but:
User A see all processes.

### PR DESCRIPTION
**Pull request by lciolecki**
This PR closes #121 it's connected with old PR: https://github.com/TouK/nussknacker/pull/146

* Remove Permission.Admin
* Add isAdmin property at user config file
* Fetching systemCategories from processDefinition elements
* Add security for changing category, now this can do only admin
* Add some todos
* Some tests refactor

### CODE REVIEW

File: ui/server/src/main/scala/pl/touk/nussknacker/ui/NussknackerApp.scala [hunk 1]
```diff
@@ -91,6 +91,9 @@ object NussknackerApp extends App with Directives with LazyLogging {
     val typeToConfig = ProcessingTypeDeps(config, featureTogglesConfig.standaloneMode)
 
     val modelData = typeToConfig.mapValues(_.modelData)
+
+    val systemCategories = modelData.values.flatMap(_.processDefinitionCategories).toList.distinct
```
Comment (mproch) [inline ui/server/src/main/scala/pl/touk/nussknacker/ui/NussknackerApp.scala]: let's takie it from categoriesConfig (ProcessTypesForCategories) - this is easier to obtain and more 'guaranteed' to be a full list

File: ui/server/src/main/scala/pl/touk/nussknacker/ui/process/ProcessTypesForCategories.scala [hunk 2]
```diff
@@ -20,4 +20,7 @@ class ProcessTypesForCategories(config: Config) {
     categoriesToTypesMap.get(category)
   }
 
+  def getSystemCategories(): List[String] = {
```
Comment (mproch) [inline ui/server/src/main/scala/pl/touk/nussknacker/ui/process/ProcessTypesForCategories.scala]: why System? maybe just allCategories? And let's make it val

File: engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/definition/ProcessDefinitionExtractor.scala [hunk 3]
```diff
@@ -123,6 +123,18 @@ object ProcessDefinitionExtractor {
       ids.toList
     }
 
+    @Override
+    def categories: List[String] = {
```
Comment (mproch) [inline engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/definition/ProcessDefinitionExtractor.scala]: maybe let's remove it now as it's not used?

File: engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/ModelData.scala [hunk 4]
```diff
@@ -62,6 +62,10 @@ trait ModelData extends ConfigCreatorSignalDispatcher {
 
   lazy val validator: ProcessValidator = ProcessValidator.default(processWithObjectsDefinition, modelClassLoader.classLoader)
 
+  lazy val processDefinitionCategories: List[String] = {
```
Comment (mproch) [inline engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/ModelData.scala]: maybe let's remove it now as it's not used?

File: engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/definition/ProcessDefinitionExtractor.scala (hunk 3 above)
Comment (lciolecki) [reply engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/definition/ProcessDefinitionExtractor.scala]: Removed

File: ui/server/src/main/scala/pl/touk/nussknacker/ui/process/ProcessTypesForCategories.scala (hunk 2 above)
Comment (lciolecki) [reply ui/server/src/main/scala/pl/touk/nussknacker/ui/process/ProcessTypesForCategories.scala]: Changed - but, why not System? :)

File: ui/server/src/main/scala/pl/touk/nussknacker/ui/NussknackerApp.scala (hunk 1 above)
Comment (lciolecki) [reply ui/server/src/main/scala/pl/touk/nussknacker/ui/NussknackerApp.scala]: You are right [;

File: engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/ModelData.scala (hunk 4 above)
Comment (lciolecki) [reply engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/ModelData.scala]: Removed

File: ui/server/src/main/scala/pl/touk/nussknacker/ui/process/ProcessTypesForCategories.scala (hunk 2 above)
Comment (mproch) [reply ui/server/src/main/scala/pl/touk/nussknacker/ui/process/ProcessTypesForCategories.scala]: we don't use this word so far - and we already have *a lot* of vocabulary :)

Comment (lciolecki) [reply ui/server/src/main/scala/pl/touk/nussknacker/ui/process/ProcessTypesForCategories.scala]: Okej okej :) Now you can give me approve!

**Pull request merged**